File: problems/Entrenamiento UPC/Problemset/C/submissions/accepted/solucion.py
import sys
input = sys.stdin.readline

t = int(input())
for _ in range(t):
    n = int(input())
    row1 = input().strip()
    row2 = input().strip()
    
    # dp[i] = min repintados considerando columnas i..n-1
    dp = [0] * (n + 1)
    for i in range(n - 1, -1, -1):
        vert_cost = 0 if row1[i] == row2[i] else 1
        best = vert_cost + dp[i + 1]
        
        if i + 1 < n:
            horiz_cost = (0 if row1[i] == row1[i+1] else 1) + (0 if row2[i] == row2[i+1] else 1)
            best = min(best, horiz_cost + dp[i + 2])
        
        dp[i] = best
    
    print(dp[0])

File: problems/Entrenamiento UPC/Problemset/C/submissions/accepted/solucion.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        string row1, row2;
        cin >> row1 >> row2;
        
        // dp[i] = min repintados considerando columnas i..n-1
        // Opcion 1: emparejar columna i verticalmente
        // Opcion 2: emparejar columnas i,i+1 horizontalmente
        
        vector<int> dp(n + 1, 0);
        for (int i = n - 1; i >= 0; i--) {
            int vert_cost = (row1[i] != row2[i]) ? 1 : 0;
            int best = vert_cost + dp[i + 1];
            
            if (i + 1 < n) {
                int horiz_cost = (row1[i] != row1[i+1] ? 1 : 0) + (row2[i] != row2[i+1] ? 1 : 0);
                best = min(best, horiz_cost + dp[i + 2]);
            }
            
            dp[i] = best;
        }
        
        cout << dp[0] << "\n";
    }
    return 0;
}
